### ISSUE TITLE
`no-identical-title` false positive inside programmatically defined describe blocks

### ISSUE THREAD

**Issue by golergka** [jest-community/eslint-plugin-jest#691]
This is related to #107 (which was my first google result on this issue).

I have a lot of parametrized tests, which I defined like this:

```Typescript
function describeForParameter(param: string, description: (stuff: SomeStuff) => void) {
    const someStuff = createSomeStuff(param)
    describe(`param is ${param}`, description(param))
}
```

I usually use this to construct objects I later use. Inside those blocks, I may have describe blocks with the same titles:

```Typescript
describeForParameter('value1', (stuff) => {
    describe('method()', () => { /* ... */ })
})
describeForParameter('value2', (stuff) => {
    describe('method()', () => { /* ... */ })
})
```

The full name would be `param is value1 > method()` in fist case, and `param is value2 > method()` in the second case, so the titles are not really identical; however, linter is not currently able to understand this.

**Comment by G-Rath** [comment]
You're effectively aliasing describe, which we don't support due to complexity reasons.

While I think the rule could be modified to restart its title count if it comes across a non-describe function call, we'd be assuming the function is an alias of `describe` which it might not be.

I'm leaning towards saying "won't fix", as I think the extra complexity and possible false positives aren't worth the small gains, given that you can use `eslint-disable`, wrap the `describeForParameter` calls in their own `describes`, or just inline it into each `describe`. You could also use `describe.each` if the tests are the same.

@SimenB do you have any thoughts on this?

**Comment by G-Rath** [comment]
I'm going to close this for the reasons explained in my last comment.

**Comment by G-Rath** [comment]
I'm going to close this because I don't think it's a bug: the idea with this rule is that it's from the perspective of the test runner whose output is only affected by `describe`, `test`, and `it` calls.

So while in the code the `describe`s are in different blocks, the test results won't reflect that if you don't have a `describe` in your `describeForParameter`.

My previous comment outlines easy ways around this, and also why it's not something that'll be worth us trying to detect.
